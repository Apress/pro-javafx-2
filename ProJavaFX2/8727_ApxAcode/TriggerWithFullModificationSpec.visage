/*
 * @test
 * @run
 */

println("Initializing seq...");
var seq = [1, 3, 5, 7, 9] on replace oldValue[lo..hi] = newSlice {
  print("  Variable seq changed from ");
  print(oldValue);
  print(" to ");
  println(seq);
  println("  The change occurred at low indexe {lo} and high index {hi}.");
  print("  The new slice is ");
  println(newSlice);
}

println('Executing "insert [2, 4, 6] before seq[3]"...');
insert [2, 4, 6] before seq[3];

println('Executing "delete seq [1..2]"...');
delete seq [1..2];

println('Executing "seq[4..5] = [8, 10]"...');
seq[4..5] = [8, 10];
