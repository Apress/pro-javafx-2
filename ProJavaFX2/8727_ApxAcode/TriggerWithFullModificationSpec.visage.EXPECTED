Initializing seq...
  Variable seq changed from [ ] to [ 1, 3, 5, 7, 9 ]
  The change occurred at low indexe 0 and high index -1.
  The new slice is [ 1, 3, 5, 7, 9 ]
Executing "insert [2, 4, 6] before seq[3]"...
  Variable seq changed from [ 1, 3, 5, 7, 9 ] to [ 1, 3, 5, 2, 4, 6, 7, 9 ]
  The change occurred at low indexe 3 and high index 2.
  The new slice is [ 2, 4, 6 ]
Executing "delete seq [1..2]"...
  Variable seq changed from [ 1, 3, 5, 2, 4, 6, 7, 9 ] to [ 1, 2, 4, 6, 7, 9 ]
  The change occurred at low indexe 1 and high index 2.
  The new slice is [ ]
Executing "seq[4..5] = [8, 10]"...
  Variable seq changed from [ 1, 2, 4, 6, 7, 9 ] to [ 1, 2, 4, 6, 8, 10 ]
  The change occurred at low indexe 4 and high index 5.
  The new slice is [ 8, 10 ]
